litmus-portal/graphql-server/pkg/chaos-workflow/service.go

```diff
@@ -408,6 +409,26 @@ func processWorkflowManifest(workflow *model.ChaosWorkFlowRequest, weights map[s
 		}
 	}
 
+	return newWeights, nil
+}
+
+func processWorkflowManifest(workflow *model.ChaosWorkFlowRequest, weights map[string]int) error {
+	var (
+		newWeights       []*model.WeightagesInput
+		workflowManifest v1alpha1.Workflow
+	)
+	err := json.Unmarshal([]byte(workflow.WorkflowManifest), &workflowManifest)
+	if err != nil {
+		return errors.New("failed to unmarshal workflow manifest")
+	}
+
+	workflowManifest.Labels = updateManifestLabels(workflowManifest.Labels, *workflow.WorkflowID, workflow.ClusterID, false)
+
+	newWeights, err = generateWeightsFromManifest(workflowManifest.Spec.Templates, workflowManifest.Labels, weights)
```

Comment:
Can you check if the weights are added in workflow manifest as well. I can see that you are returning the weight list in this function, these weights should be added in chaos engine manifest as well.

Reply:
I feel the weights are not being added to the main manifest with this logic because although I am adding it, it is restricted to the function itself. I will return the manifest as well in the function and then set it to the main manifest when calling the function in my next commit.

Reply:
@amityt
I have made the required changes in my latest commit.